src/sentry/integrations/repository/issue_alert.py

```diff
@@ -140,3 +151,32 @@ def create_notification_message(
                 extra=data.__dict__,
             )
             raise
+
+    def get_all_parent_notification_messages_by_filters(
+        self, group_ids: list[int] | None = None, project_ids: list[int] | None = None
+    ) -> Generator[IssueAlertNotificationMessage, None, None]:
+        """
+        If no filters are passed, then all parent notification objects are returned.
```

Comment:
Is there any place in your specs that we would call this with no filter? If not, how about making `(project_ids or group_ids)` required?

Reply:
Not in the flows for this feature, but I think requiring either/or is too stringent/specific, and doesn't leave possibility for the future as we add additional filters

Reply:
Reading your comment I thought you're worried about someone calling this with no filter :P which is why I asked.

Reply:
oh haha, no not at all! Because it can return a lot, with or without filters, I've ensured it uses a generator

Reply:
I updated the wording of the comment, hopefully this removes that implication! Users should feel free to get all the data they want